query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -323,34 +331,31 @@ private String convertExpressionToMapValueColumn(Expression expression) {
     throw new IllegalArgumentException("operator supports multi value column only");
   }
 
-  private LiteralConstant[] convertExpressionToMapLiterals(Expression expression) {
-    LiteralConstant[] literals = new LiteralConstant[2];
+  private List<LiteralConstant> convertExpressionToMapLiterals(Expression expression) {
+    List<String> literals = new ArrayList<>(List.of("", ""));
     if (expression.getValueCase() == LITERAL) {
```

Comment:
Also, do we throw an exception here if rhs is not literal ? As using the below design, we always expect the key. Also, we are always calling this method from such a case so it may prevent any error if some new case is added. 

```
@Value class MapKeyValueArguments { LiteralConstant key; Optional<LiteralConstant> value; }
```

Reply:
CC @kotharironak

Reply:
In the existing `main` code, we are defaulting to empty string `""` if the case is not `LITERAL`. Here as well default to `""` string LiteralConstant.

Reply:
I thought optional was to deal with this default ""

Reply:
Optional<LiteralConstant> value means that the map's value argument is not mandatory. Refer to this - https://github.com/hypertrace/query-service/pull/125#discussion_r770359421

Reply:
So, in a way, we can also extend as you suggested that we expect the `key` argument should be present. if it's not there, we can throw the exception.

Does the below make sense?
`spanTags contains_key ''`